src/app/public/modules/affix/affix.directive.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnDestroy,
+  Output,
+  SimpleChanges
+} from '@angular/core';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  SkyAffixHorizontalAlignment
+} from './affix-horizontal-alignment';
+
+import {
+  SkyAffixPlacement
+} from './affix-placement';
+
+import {
+  SkyAffixSubjectVisibilityChange
+} from './affix-subject-visibility-change';
+
+import {
+  SkyAffixVerticalAlignment
+} from './affix-vertical-alignment';
+
+import {
+  SkyAffixService
+} from './affix.service';
+
+import {
+  SkyAffixer
+} from './affixer';
+
+/**
+ * Affixes the host element to a target element.
+ */
+@Directive({
+  selector: '[skyAffixTo]'
+})
+export class SkyAffixDirective implements OnChanges, OnDestroy {
+
+  /**
+   * The target element to affix the host element.
+   */
+  @Input()
+  public skyAffixTo: HTMLElement;
```

Comment:
I'd prefer to keep this`Input` at the top of the class since it's the primary selector. Thoughts?

Reply:
Yeah, I think for directives, that makes sense 👍